src/OrleansProviders/Streams/Memory/MemoryMessageData.cs

```diff
@@ -0,0 +1,39 @@
+﻿
+using System;
+
+namespace Orleans.Providers
+{
+    /// <summary>
+    /// Represents the event sent and received from an In-Memory queue grain. 
+    /// </summary>
+    [Serializable]
+    public struct MemoryMessageData
+    {
+        /// <summary>
+        /// Stream Guid of the event data.
+        /// </summary>
+        public Guid StreamGuid;
+        /// <summary>
+        /// Hash code of namespace. Stores hash for fast and efficient storage.
+        /// </summary>
+        public string StreamNamespace;
```

Comment:
Since it's not actually using the hash of namespace here, the comment should be changed. However, I think it's reasonable to consider using Hash since MemoryMessageData is also used as cache structure.